From 457d0d0a6c6d816745db0c5127c437a82b07424d Mon Sep 17 00:00:00 2001
From: Patrick Mooney <pmooney@pfmooney.com>
Date: Thu, 15 Dec 2016 21:24:54 +0000
Subject: [PATCH] OS-5850 kmdb_fault() continuously prints that it's trying to
 enter safe mode Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
 Reviewed by: Robert Mustacchi <rm@joyent.com> Approved by: Robert Mustacchi
 <rm@joyent.com>

---
 usr/src/cmd/mdb/common/kmdb/kmdb_fault.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/usr/src/cmd/mdb/common/kmdb/kmdb_fault.c b/usr/src/cmd/mdb/common/kmdb/kmdb_fault.c
index 06f7c2927f..4698f3cc84 100644
--- a/usr/src/cmd/mdb/common/kmdb/kmdb_fault.c
+++ b/usr/src/cmd/mdb/common/kmdb/kmdb_fault.c
@@ -22,10 +22,9 @@
 /*
  * Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
  * Use is subject to license terms.
+ * Copyright 2016 Joyent, Inc.
  */
 
-#pragma ident	"%Z%%M%	%I%	%E% SMI"
-
 /*
  * Handling of unintentional faults (i.e. bugs) in the debugger.
  */
@@ -44,7 +43,8 @@
 void
 kmdb_fault(kreg_t tt, kreg_t pc, kreg_t sp, int cpuid)
 {
-	int debug_self_confirm = 0, try;
+	int debug_self_confirm = 0;
+	volatile int try;
 	jmp_buf pcb, *old;
 	char c;
 
-- 
2.21.0

